website/static/js/tests/registrationUtils.test.js

```diff
@@ -0,0 +1,448 @@
+            assert.equal(q.id, id);
+            assert.equal(q.title, question.title);
+            assert.equal(q.nav, question.nav);
+            assert.equal(q.type, question.type);
+            assert.equal(q.format, question.format);
+            assert.equal(q.description, question.description);
+            assert.equal(q.help, question.help);
+            assert.equal(q.required, question.required);
+            assert.equal(q.options, question.options);
+            assert.isDefined(q.value);
+        });
+    });
+    describe('#isComplete', () => {
+        it('is true if the Question\'s value is not blank', () => {
+            assert.isFalse(q.isComplete());
+            q.value('not blank');
+            assert.isTrue(q.isComplete());
+        });
+    });
+    describe('#isValid', () => {
+        it('is true if the Question\'s value is not empty and the question is required', () => {
+            assert.isFalse(q.value.isValid());
+            q.value('not empty');
+            assert.isTrue(q.value.isValid());
+        });
+    });
+    describe('#init', () => {
+        it('maps object-type Question\'s properties property to sub-Questions', () => {
+            var props = {
+                foo: {
+                    type: 'number'
+                }
+            };
+
+            var objType = new Question({
+                type: 'object',
+                properties: props
+            });
+            var obj = new Question(objType);
+            assert.equal(obj.properties.foo.id, 'foo');
+            assert.isDefined(obj.properties.foo.value);
+        });
+    });
+    describe('#toggleExample', () => {
+        it('toggles the value of Question.showExample', () => {
+            assert.isFalse(q.showExample());
+            q.toggleExample();
+            assert.isTrue(q.showExample());
+        });
+    });
+});
+
+describe('MetaSchema', () => {
+    describe('#constructor', () => {
+        it('loads optional instantion data and maps question data to Question instances', () => {
+
+            var ctx = mkMetaSchema();
+            var qid = ctx[0];
+            var params = ctx[1];
+            var ms = ctx[2];
+            assert.equal(ms.name, params.schema_name);
+            assert.equal(ms.version, params.schema_version);
+            assert.equal(ms.schema.pages[0].id, params.schema.pages[0].id);
+
+            assert.isDefined(ms.schema.pages[2].questions[qid].value);
+        });
+    });
+    describe('#flatQuestions', () => {
+        it('creates a flat array of the schema questions', () => {
+            var ctx = mkMetaSchema();
+            var qid = ctx[0];
+            var params = ctx[1];
+            var ms = ctx[2];
+
+            var questions = [];
+            $.each(params.schema.pages, function(i, page) {
+                $.each(page.questions, function(qid, question) {
+                    questions.push(question);
+                });
+            });
```

Comment:
Can you rework the test to check the result directly? The logic used to generate `var questions` is extremely similar to the logic in `flatQuestions`, so it would be very surprising if this test ever fails.